bundles/automation/org.eclipse.smarthome.automation.core/src/main/java/org/eclipse/smarthome/automation/core/internal/RuleEngineImpl.java

```diff
@@ -59,135 +58,293 @@
+     * Posts an event through the event bus in an asynchronous way. {@link RuleEngineImpl} use it for posting the
+     * {@link RuleStatusInfoEvent}.
+     */
+    private EventPublisher eventPublisher;
+    private static final String SOURCE = RuleEngineImpl.class.getSimpleName();
+
+    private final ModuleHandlerCallback moduleHandlerCallback = new ModuleHandlerCallback() {
+
+        @Override
+        public Boolean isEnabled(String ruleUID) {
+            return RuleEngineImpl.this.isEnabled(ruleUID);
+        }
+
+        @Override
+        public void setEnabled(String uid, boolean isEnabled) {
+            RuleEngineImpl.this.setEnabled(uid, isEnabled);
+        }
+
+        @Override
+        public RuleStatusInfo getStatusInfo(String ruleUID) {
+            return RuleEngineImpl.this.getStatusInfo(ruleUID);
+        }
+
+        @Override
+        public RuleStatus getStatus(String ruleUID) {
+            return RuleEngineImpl.this.getStatus(ruleUID);
+        }
+
+        @Override
+        public void runNow(String uid) {
+            RuleEngineImpl.this.runNow(uid);
+        }
+
+        @Override
+        public void runNow(String uid, boolean considerConditions, Map<String, Object> context) {
+            RuleEngineImpl.this.runNow(uid, considerConditions, context);
+        }
+
+    };
+
+    /**
+     * Constructor of {@link RuleEngineImpl}. It initializes the logger and starts tracker for
+     * {@link ModuleHandlerFactory}
      * services.
-     *
-     * @param bc {@link BundleContext} used for tracker registration and rule engine logger creation.
-     * @throws InvalidSyntaxException
      */
-    public RuleEngine() {
-        this.rules = new HashMap<String, RuntimeRule>(20);
+    public RuleEngineImpl() {
         this.contextMap = new HashMap<String, Map<String, Object>>();
         this.moduleHandlerFactories = new HashMap<String, ModuleHandlerFactory>(20);
     }
 
+    /**
+     * This method is used to create a {@link CompositeModuleHandlerFactory} that handles all composite
+     * {@link ModuleType}s. Called from DS to activate the rule engine component.
+     *
+     * @param context a {@link BundleContext} used for {@link CompositeModuleHandlerFactory} creation.
+     */
+    @Activate
+    protected void activate(BundleContext context) {
+        compositeFactory = new CompositeModuleHandlerFactory(context, mtRegistry, this);
+
+        // enable the rules that are not persisted as Disabled;
+        for (Rule rule : ruleRegistry.getAll()) {
+            String uid = rule.getUID();
+            if (disabledRulesStorage == null || disabledRulesStorage.get(uid) == null) {
+                setEnabled(uid, true);
+            }
+        }
+    }
+
+    /**
+     * Bind the {@link ModuleTypeRegistry} service - called from DS.
+     *
+     * @param moduleTypeRegistry a {@link ModuleTypeRegistry} service.
+     */
+    @Reference(cardinality = ReferenceCardinality.MANDATORY, policy = ReferencePolicy.STATIC)
```

Comment:
mandatory and static are the default, in other PRs we told the contributors to remove that statements.